travel-agency/public/js/ajax-city-handler.js

```diff
@@ -0,0 +1,98 @@
+$.ajaxSetup({
+    headers: {
+        'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+    }
+});
+$('form').submit(function(event){
+    event.preventDefault();
+})
+function createCity(){
+    var currentPage = new URLSearchParams(window.location.search).get('page') || 1;
+    var jsonData = formToJson($('#add-city-form'),currentPage);
+
+    $.ajax({
+        url: '/',
+        method: 'POST',
+        data: JSON.stringify(jsonData),
+        dataType: 'JSON',
+        contentType: 'application/json',
+        cache: false,
+        processData: false,
+        success:function(response)
+        {
+            document.forms["add-city-form"].reset();
+            $('#name-error').text('');
+            $('#cities-table').html(response.updatedCitiesTable)
+            $('#pagination-links').html(response.updatedPaginationLinks);
+        },
+        error: function(xhr) {
+            if (xhr.status === 422) {
+                var errors = xhr.responseJSON.errors;
+                $('#name-error').text(errors.name[0]);
+            }
+            console.error(xhr.responseText)
+        }
+    });
+}
+
+function deleteCity(cityId){
+    var currentPage = new URLSearchParams(window.location.search).get('page') || 1;
+    var jsonData = {'id': cityId,
+                    'page': currentPage};
+
+    $.ajax({
+        url: '/',
+        method: 'DELETE',
+        data: JSON.stringify(jsonData),
+        dataType: 'JSON',
+        contentType: 'application/json',
+        cache: false,
+        processData: false,
+        success:function(response)
+        {
+            $('#cities-table').html(response.updatedCitiesTable)
+            $('#pagination-links').html(response.updatedPaginationLinks);
+        },
+        error: function(xhr) {
+            if (xhr.status === 422) {
```

Comment:
Same thing as before about what would happen if the HTTP status code was something other than `422`.